src/sessions/__tests__/SessionManager.test.ts

```diff
@@ -773,4 +773,23 @@ describe('SessionManager tests', () => {
 
         expect(actualSessionAttributes.domain).toEqual('overridden');
     });
+
+    test('when session is created then it will use global sampling decision', async () => {
+        const sessionManager = defaultSessionManager({
+            ...DEFAULT_CONFIG,
+            sessionSampleRate: 0.5
+        });
+
+        // This test is not foolproof. It will give a false positive 1% of the time.
```

Comment:
issue: If needed after the next revision, mock the random value generator to get rid of this source of flakiness.

Reply:
Thanks for pointing this out! Used this to remove flakiness.